### ISSUE TITLE
directives not attaching to heredocs

### ISSUE THREAD

**Issue by mmlb** [koalaman/shellcheck#1071]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1117
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

foo=foo

#shellcheck disable=SC1117
cat >>/tmp/foo <<-EOF
hello\tworld
EOF

echo foo=${foo:-}

```
If you comment out the `foo=foo` then the disable directive applies file wide and shellcheck then ignores the `\t`.

#### Here's what shellcheck currently says:

```
Line 7:
hello\tworld
     ^-- SC1117: Backslash is literal in "\t". Prefer explicit escaping: "\\t".
```

#### Here's what I wanted or expected to see:

no output

**Comment by mmlb** [comment]
Thanks for all the work on shellcheck!
